src/data_gathering.py

```diff
@@ -146,22 +227,38 @@ def process_block(block_number, bundles):
         print(f"Failed to process block {block_number}: {e}")
 
 if __name__ == "__main__":
-    # Define the block range or specific blocks to process from config
-    latest_block_number = web3.eth.block_number
-    start_block_offset = config.get('start_block_offset', 100)  # How far back to start from the latest block
-    num_blocks_to_process = config.get('num_blocks_to_process', 5)  # How many blocks to process
+    # Fetch MEV Blocker bundles
+    bundles = get_mev_blocker_bundles()
 
-    start_block = latest_block_number - start_block_offset
-    end_block = latest_block_number
-
-    # Fetch MEV Blocker bundles once
-    bundles = get_mev_blocker_bundles(start_block, end_block)
+    # Option, which handles logic when 0 results are returned
+    abort_on_empty_first_query = config.get('abort_on_empty_first_query', True)
 
     if not bundles:
-        print("No bundles retrieved. Exiting the script.")
+        if abort_on_empty_first_query:
+            print("No bundles retrieved. Exiting the script.")
+            exit(1)
+        else:
+            print("No bundles retrieved. Proceeding to the next query...")
+
+    # Fetch non-MEV transactions
+    non_mev_transactions = get_non_mev_transactions()
+
+    if not non_mev_transactions:
+        print("No non-MEV transactions retrieved. Exiting the script.")
         exit(1)
 
-    block_numbers = [latest_block_number - i for i in range(num_blocks_to_process)]  # Process the configured number of blocks
+    # Determine the number of blocks to process
+    latest_block_number = web3.eth.block_number
+    latest_processed_block = get_latest_processed_block() or latest_block_number - config.get('start_block_offset', 100)
+
+    # Check if `num_blocks_to_process` is set to "all"
+    num_blocks_to_process = config.get('num_blocks_to_process', 5)
+    if num_blocks_to_process == "all":
```

Comment:
This logic is quite implicit and not really intuitive (fetch last 100 blocks in intervals of 5 when nothing is specified). Can we make sure we log the range for which the script is running so that this is easily observable in the output?

Also, I think it should be possible to specify a specific start block somehow (in case there has been a big gap we want to backfill).

maybe just having start block (with default last processed if found, otherwise current-block 100), end block (with default current block) and chunk size (default whatever is reasonable with a free dune account) makes sense?